src/codegen/compile.ml

```diff
@@ -7102,7 +7104,9 @@ module MakeSerialization (Strm : Stream) = struct
         let (set_x, get_x) = new_local env "x" in
         let (set_val, get_val) = new_local env "val" in
         let (set_arg_typ, get_arg_typ) = new_local env "arg_typ" in
-        with_composite_typ idl_vec (ReadBuf.read_sleb128 env) ^^ set_arg_typ ^^
+        with_composite_typ idl_vec (fun get_typ_buf ->
+        ReadBuf.read_sleb128 env get_typ_buf ^^
+        set_arg_typ ^^
```

Comment:
Looks reasonable. Maybe indent the code now inside the `(`?

It seems that array is using `remember_failure`, so the full array is deserialized, even if there are failures. That’s good.

Reply:
(now indented)